modules/web/src/app/node-data/component.ts

```diff
@@ -295,34 +296,21 @@ export class NodeDataComponent extends BaseFormValidator implements OnInit, OnDe
         this.setDefaultOperatingSystemProfiles();
       });
 
-    this._settingsService.adminSettings.pipe(take(1)).subscribe(settings => {
-      this.allowedOperatingSystems = settings?.allowedOperatingSystems && settings.allowedOperatingSystems;
-      this._setDefaultOS();
-
-      const autoUpdatesEnabled = settings.machineDeploymentOptions.autoUpdatesEnabled;
-      const replicas =
-        this.dialogEditMode || this.isCusterTemplateEditMode
-          ? this._nodeDataService.nodeData.count
-          : settings.defaultNodeCount;
-
-      this.form.get(Controls.Count).setValue(replicas);
-      if (this.dialogEditMode) {
-        if (autoUpdatesEnabled && settings.machineDeploymentOptions.autoUpdatesEnforced) {
-          if (!this.form.get(Controls.UpgradeOnBoot).value) {
-            this.form.get(Controls.UpgradeOnBoot).setValue(autoUpdatesEnabled);
-          }
-          if (this.form.get(Controls.DisableAutoUpdate).value) {
-            this.form.get(Controls.DisableAutoUpdate).setValue(!autoUpdatesEnabled);
-          }
-        }
-      } else {
-        this.form.get(Controls.UpgradeOnBoot).setValue(autoUpdatesEnabled);
-        this.form.get(Controls.DisableAutoUpdate).setValue(!autoUpdatesEnabled);
-      }
-      if (settings.machineDeploymentOptions.autoUpdatesEnforced) {
-        this.form.get(Controls.UpgradeOnBoot).disable();
-        this.form.get(Controls.DisableAutoUpdate).disable();
+    forkJoin([
+      this._settingsService.adminSettings.pipe(take(1)),
+      this._projectService.selectedProject.pipe(take(1)),
+    ]).subscribe(([settings, project]) => {
```

Comment:
We can apply `take(1)` on the `forkJoin`.

```suggestion
    forkJoin([
      this._settingsService.adminSettings,
      this._projectService.selectedProject,
    ]).pipe(take(1)).subscribe(([settings, project]) => {
```

Reply:
Unfortunately, that didn't work.